### PR TITLE
#1207 Updated to not display units when missing durations 

### PR DESCRIPTION
**Pull request by khatril**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by thomas-neitmann** [comment]
Thanks a lot @khatril! Could you please add a unit test for this feature. Also there's a coupe of workflows failing. Please take a look.

**Comment by thomas-neitmann** [comment]
Please add to the existing unit test file.

File: NEWS.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ admiral extension packages (#1079)
 
 - Randomization Date `RANDDT` variable added to ADSL template and vignette (#1126)
 
+- `derive_vars_duration()` Updated to not display units when there is missing duration (#1207)
```
Comment (thomas-neitmann) [inline NEWS.md:12]: Please make "Updated" lowercase.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]
Just a minor thing before we can merge 😄

**Comment by thomas-neitmann** [comment]
Also some workflows are still failing.

**Comment by khatril** [comment]
@thomas-neitmann Not sure how to resolve the 'Check URLS' ?

**Comment by bms63** [comment]
image

I think the Network Error is fine to ignore.  I don't think it is on our end.

File: R/derive_vars_duration.R [hunk 2]
```diff
@@ -143,7 +146,7 @@ derive_vars_duration <- function(dataset,
     )
 
   if (!quo_is_null(new_var_unit)) {
-    dataset <- dataset %>% mutate(!!new_var_unit := toupper(out_unit))
+    dataset <- dataset %>% mutate(!!new_var_unit := ifelse(is.na(!!new_var), NA, toupper(out_unit)))
```
Comment (thomas-neitmann) [inline R/derive_vars_duration.R:149]: Please use `if_else` and make sure to replace `NA` with `NA_character_`.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

File: R/derive_vars_duration.R [hunk 3]
```diff
@@ -88,7 +88,10 @@
 #' @examples
 #' data <- tibble::tribble(
 #'   ~BRTHDT, ~RANDDT,
-#'   lubridate::ymd("1984-09-06"), lubridate::ymd("2020-02-24")
+#'   lubridate::ymd("1984-09-06"), lubridate::ymd("2020-02-24"),
+#'   lubridate::ymd("1985-01-01"), NA,
+#'   NA, lubridate::ymd("2021-03-10"),
+#'   NA, NA
```
[the remark this answers is not in the document]
Comment (vrakinav) [reply R/derive_vars_duration.R]: Updated

File: R/derive_vars_duration.R (hunk 2 above)
Comment (vrakinav) [reply R/derive_vars_duration.R]: Updated

File: tests/testthat/test-derive_vars_duration.R [hunk 4]
```diff
@@ -1,9 +1,12 @@
 test_that("duration and unit variable are added", {
   input <- tibble::tribble(
     ~BRTHDT, ~RANDDT,
-    ymd("1999-09-09"), ymd("2020-02-20")
+    lubridate::ymd("1984-09-06"), lubridate::ymd("2020-02-24"),
+    lubridate::ymd("1985-01-01"), NA,
+    NA, lubridate::ymd("2021-03-10"),
+    NA, NA
```
[the remark this answers is not in the document]
Comment (vrakinav) [reply tests/testthat/test-derive_vars_duration.R]: Updated

**Comment by vrakinav** [comment]
Hi @thomas-neitmann I have updated files according to your comments. Could you please check them when you have time?

Thanks,
Vira

**Review by thomas-neitmann — APPROVED** [review]
Thanks a lot!

**Pull request merged**